doc/source/data/glossary.rst

```diff
@@ -0,0 +1,140 @@
+.. _datasets_glossary:
+
+=====================
+Ray Datasets Glossary
+=====================
+
+.. glossary::
+
+    Batch format
+        The way batches of data are represented.
+
+        Set ``batch_format`` in methods like :meth:`~ray.data.Dataset.iter_batches` and
+        :meth:`~ray.data.Dataset.map_batches` to specify the batch type.
+
+        .. doctest::
+
+            >>> dataset = ray.data.range_table(10)
+            >>> next(iter(dataset.iter_batches(batch_format="numpy", batch_size=5)))
+            {'value': array([0, 1, 2, 3, 4])}
+            >>> next(iter(dataset.iter_batches(batch_format="pandas", batch_size=5)))
```

Comment:
shall we also add an example for `map_batches(batch_format=...)`?

Reply:
I don't think we should. Meaningful `map_batches` examples are relatively complicated, and I don't think we should include long examples in the glossary. Also, since we link to `UDF Input Batch Formats`, I think we should be okay.